content/Rust-1.87.0.md

```diff
@@ -0,0 +1,252 @@
++++
+path = "2025/05/15/Rust-1.87.0"
+title = "Announcing Rust 1.87.0"
+authors = ["The Rust Release Team"]
+aliases = [
+    "2025/05/15/Rust-1.87.0.html",
+    "releases/1.87.0",
+]
+
+[extra]
+release = true
++++
+
+The Rust team is happy to announce a new version of Rust, 1.87.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.87.0 with:
+
+```
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.87.0](https://doc.rust-lang.org/stable/releases.html#version-1870-2025-04-03).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.87.0 stable
+
+### Anonymous pipes
+
+1.87 adds access to anonymous pipes to the standard library. This includes
+integration with `std::process::Command`'s input/output methods.
+
+As a toy example, this program uses pipes to stream data into `cat` and read the output:
+
+```rust
+use std::process::Command;
+use std::io::{pipe, Read, Write};
+
+let (ping_rx, mut ping_tx) = pipe()?;
+let (mut pong_rx, pong_tx) = pipe()?;
+
+// Spawn a process that echoes its input.
+let mut echo_server = Command::new("cat").stdin(ping_rx).stdout(pong_tx).spawn()?;
+
+ping_tx.write_all(b"hello")?;
+// Close to unblock echo_server's reader.
+drop(ping_tx);
+
+let mut buf = String::new();
+// Block until echo_server's writer is closed.
+pong_rx.read_to_string(&mut buf)?;
+assert_eq!(&buf, "hello");
+
+echo_server.wait()?;
+```
+
+### Safe architecture intrinsics
+
+Most `std::arch` intrinsics that are unsafe only due to requiring target
+features to be enabled are now callable in safe code that has those features
+enabled. For example, the following toy program which implements summing an array using
+manual intrinsics can now use safe code for the core loop.
+
+```rust
+#![forbid(unsafe_op_in_unsafe_fn)]
+
+use std::arch::x86_64::*;
+
+fn sum(slice: &[u32]) -> u32 {
+    #[cfg(target_arch = "x86_64")]
+    {
+        if is_x86_feature_detected!("avx2") {
+            // SAFETY: We have detected the feature is enabled at runtime,
+            // so it's safe to call this function.
+            return unsafe { sum_avx2(slice) };
+        }
+    }
+
+    slice.iter().sum()
+}
+
+#[target_feature(enable = "avx2")]
+#[cfg(target_arch = "x86_64")]
+fn sum_avx2(slice: &[u32]) -> u32 {
```

Comment:
Maybe there's a better (shorter?) example that can be fully safe code now except the function call? Something like crc32 perhaps that operates on just u64 values and so doesn't need the align/transmutes?